app/models/post.rb

```diff
@@ -0,0 +1,23 @@
+class Post < ApplicationRecord
+  # Establishes a belongs_to association with the User model, using author_id as the foreign key.
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  # Establishes has_many associations with the Comment and Like models.
+  has_many :comments
+  has_many :likes
+
+  # After saving a post, this callback method updates the author's posts counter.
+  after_save :update_author_posts_counter
+
+  # Returns the 'limit' most recent comments for a post, defaults to 5.
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  private
+
+  # Callback method to update the author's posts counter.
+  def update_author_posts_counter
+    author.increment!(:posts_count)
+  end
```

Comment:
- [ ] Kindly add validations for the posts model as well as requested by the previous reviewer. You have to meet these requirements if you want to move to the next milestone.
- [ ] Kindly add specs for the validations you will create, and the model methods you created in the previous milestone.


![model](https://github.com/kayjeee/Kayjeee_Postgresql_blogapp/assets/53124548/6bb2c6bf-c3fa-4f06-965a-8c32bfb3fc5e)


![specs](https://github.com/kayjeee/Kayjeee_Postgresql_blogapp/assets/53124548/b7963776-508d-4fd7-9ae5-19f5ee45addb)